api/common/modelstatus.go

```diff
@@ -37,60 +43,75 @@ func (c *ModelStatusAPI) ModelStatus(tags ...names.ModelTag) ([]base.ModelStatus
 		return nil, err
 	}
 
-	results := make([]base.ModelStatus, len(result.Results))
-	for i, r := range result.Results {
+	return c.processModelStatusResults(result.Results)
+}
+
+func (c *ModelStatusAPI) processModelStatusResults(rs []params.ModelStatus) ([]base.ModelStatus, error) {
+	results := make([]base.ModelStatus, len(rs))
+	for i, r := range rs {
+		if r.Error != nil {
+			// cope with typed error
+			results[i].Error = errors.Trace(r.Error)
+			continue
+		}
 		model, err := names.ParseModelTag(r.ModelTag)
 		if err != nil {
-			return nil, errors.Annotatef(err, "ModelTag %q at position %d", r.ModelTag, i)
+			results[i].Error = err
```

Comment:
I'd suggest adding a Trace to this error, and below so it's clear that the error has come from here if it ever happens.